### ISSUE TITLE
Faster testing and playing for 99% players

### ISSUE THREAD

**Issue by Elvenisko** [wesnoth/wesnoth#7059]
### Describe the desired feature

When starting any multiplayer map, I have to change every player from "human" to "AI". It is slow process for maps with multiple players. I believe it is great to make "human" only player 1 and all others make an AI controlled player.

This will make everything much faster, because only from time to time does 2 or more players play on single computer. So, 99% times it is better to make everyone except player 1 an AI player.

**Comment by knyghtmare** [comment]
The MP system has been meant for players playing with other players and only switching sides to AI when no other players can join within the waiting time. So, this request seems rather irrelevant to me. You may make changes to the local copies of your client to default to an AI side though.

**Comment by soliton-** [comment]
I assume this request is about playing a local multiplayer game on a custom map with no scenario file involved.

**Comment by Elvenisko** [comment]
Soliton is right. This is for anyone, who wish to play against AI and have to switch everything everytime.

Also for guys, who wish to try their new map made in editor against AI players.

Also for everyone, who wish for some reason (testing map, new faction etc) view a game between only AI players. If someone wish to make 20 tests, he/she have to 20x change irrational game setting.

When playing with friend, it is very easy to take one side to Joe and another to Frank, so they can go together against AI.

Simply it is MUCH BETTER to make everyone AI in the default setting for everyone. Or add ability to change it in graphic mode.

**Comment by Wedge009** [comment]
@soliton- Isn't this about the lobby rather than MP mechanics?

**Comment by stevecotton** [comment]
> When playing with friend, it is very easy to take one side to Joe and another to Frank, so they can go together against AI.

I'm against changing the default, because of what happens when you forget to switch a side from AI - the first human player plays their turn, and then the AI kicks in at the point that the second human player would play. It goes from a short annoyance to requiring a restart of the game.

**Comment by Elvenisko** [comment]
How many times does this happen? About... Never? Compared to everytime... Or in percent, 1% against 99%...

**Comment by Elvenisko** [comment]
This is about lobby, a way to improve it. Of course there is more possibilities.

For example lobby with friends and lobby with AI players by default.

**Comment by Pentarctagon** [comment]
You can save the map as a scenario via the editor and then set the side controller to be the AI.

**Comment by Elvenisko** [comment]
> You can save the map as a scenario via the editor and then set the side controller to be the AI.

well, this is quite the opposite of what am I saying. My thinking is about making it simple, easy to use, without a lot of clicking and also well for everyday player. Many games, like for example maps from Warcraft and other games, are about playing for 2 players, even if game is for 12 games. So player have to do a looot of clicks before playing.

Okay, maybe we are looking at it the wrong way. Lets think about our possibilities.

A) do not change anything, everyone shall do what he/she want to do, even if it is much more clicking than usual
B) lets make a new default. Max number of enemies on the map, with everyone except player 1 being AI
C) like A, but add a special button to make B. Button can have text "PC vs AI players"
D) B, but with ability to make it look like A do not have a lot of sense, but still a solid option
E) add option into setting about default lobby, so everyone can choose A or B.

Thank you guys. And I am sorry if my opinions are too crazy, but thinking about it from Wesnoth 1.0 onwards and I really think this can be changed and it will be better to change it. Btw Stronghold Crusader have it this way, simply click on "add max enemies".

**Comment by Pentarctagon** [comment]
> well, this is quite the opposite of what am I saying. My thinking is about making it simple, easy to use, without a lot of clicking and also well for everyday player.

The scenario editor already allows you to do what you want to do, without changing the default for everyone else.

**Comment by Elvenisko** [comment]
I disagree. Scenario editor helps modifying scenarios. All i am thinking about is simply playing for someone who dont want to change scenarios.

Lets say you wish to try new faction on 5 different maps. Not for playing 5 times on same map.

**Comment by Pentarctagon** [comment]
If your proposal is to globally have all scenarios default to having slots be AI instead of human except for side 1, then I can just say right now that won't be implemented. Otherwise, can you clarify in exactly which situations you're asking for this change to be made?

**Comment by Elvenisko** [comment]
Well, still think it is a good idea, but if will not be implemented, than ok. No need to continue this issue.

**Comment by soliton-** [comment]
> @soliton- Isn't this about the lobby rather than MP mechanics?

To me the MP Lobby tag is for the lobby you join when you login to the MP server. I suppose you might call the pre-game setup screen a lobby as well. Especially when talking about local games that seems more confusing though.

Regarding what this is about. As far as I can tell it can only be about maps without a scenario file. As soon as you have a scenario file you must specify a controller for a side. There is no default there except perhaps a fallback for invalid WML. It would also not change how generated scenarios work.

I tend to agree that it seems very rare for two or more players to share a device and play a custom made map.

If we do not want to change the default because it might surprise those rare occasions this might also be done as an option in the create game screen which is remembered. Then there is no change for those that actually do play a custom map with multiple people on the same device and for those that want to test/play such a map against the AI they can set the option once and it will be remembered for future games.
